src/lib.rs

```diff
@@ -682,6 +732,43 @@ impl LightningNode {
         }
     }
 
+    /// Pay an LNURL-pay the provided amount.
+    ///
+    /// Parameters:
+    /// * `amount_sat` - amount to be paid
+    /// * `lnurl_pay_request_data` - LNURL-pay request data as obtained from [`LightningNode::decode_data`]
+    ///
+    /// Returns the payment hash of the payment.
+    pub fn pay_lnurlp(
+        &self,
+        amount_sat: u64,
+        lnurl_pay_request_data: LnUrlPayRequestData,
+    ) -> Result<String> {
+        let initial_latest_payments = self.get_latest_payments(1)?;
+        let initial_latest_payment = initial_latest_payments.get(0);
+
+        self.rt
```

Comment:
An `EndpointError` can be returned even though no `Payment` was created. That's why we need the hack. It's only temporary until we have the changes from https://github.com/breez/breez-sdk/pull/550.

Reply:
As you can see [here](https://github.com/breez/breez-sdk/blob/7a1d40c8bdd51b3d34ac6f3f10abd4b29b926487/libs/sdk-core/src/breez_services.rs#L294), if the second call to the LNURL service returns an error instead of an invoice, no `Payment` will be created and we will get `EndpointError`.

Reply:
Forget that, I misunderstood your comment

Reply:
I think your proposal is still hacky as it assumes, without certainty, that the last payment was the one created by the LNURL-pay flow. Given that we want to fix this the proper way anyway, that the current error model will change, both on our side as well as this specific method return, I would keep it like this and fix it when the changes to Breez get shipped.